api/app/main.py

```diff
@@ -96,20 +98,29 @@ async def get_health():
     """ A simple endpoint for Openshift Healthchecks """
     try:
         health_check = health.patroni_cluster_health_check()
+
         logger.info('/health - healthy: %s. %s',
                     health_check.get('healthy'), health_check.get('message'))
+
         return health_check
     except Exception as exception:
         logger.error(exception, exc_info=True)
         raise
 
 
 @api.post('/observations/', response_model=schemas.observations.WeatherStationHourlyReadingsResponse)
-async def get_hourlies(request: schemas.stations.StationCodeList, _: bool = Depends(authenticate)):
+async def get_hourlies(request: schemas.shared.WeatherDataRequest, _: bool = Depends(authenticate)):
     """ Returns hourly observations for the last 5 days, for the specified weather stations """
     try:
         logger.info('/observations/')
-        readings = await hourlies.get_hourly_readings(request.stations)
+
+        if request.time_of_interest is not None:
```

Comment:
See my comment in WeatherDataRequest - I think we can get rid of the 4 lines of code that are repeated a few times.